gradio/components/image_editor.py

```diff
@@ -0,0 +1,297 @@
+"""gr.Image() component."""
+
+from __future__ import annotations
+
+import warnings
+from pathlib import Path
+from typing import Any, Iterable, Literal, Optional, TypedDict, Union, cast
+
+import numpy as np
+from gradio_client import utils as client_utils
+from gradio_client.documentation import document, set_documentation_group
+from PIL import Image as _Image  # using _ to minimize namespace pollution
+
+import gradio.image_utils as image_utils
+from gradio import utils
+from gradio.components.base import Component
+from gradio.data_classes import FileData, GradioModel
+from gradio.events import Events
+
+set_documentation_group("component")
+_Image.init()  # fixes https://github.com/gradio-app/gradio/issues/2843
+
+
+class PreprocessData(TypedDict):
```

Comment:
Nice, its cool to have a typeddict to store the value as well. I find `PreprocessData` a bit confusing name, since its also the format that a user is expected to return from their function. How about `EditorValue` as it should be the same type as `value`